utils/common/src/main/java/org/apache/brooklyn/util/http/HttpExecutor.java

```diff
@@ -0,0 +1,37 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.brooklyn.util.http;
+
+import org.apache.http.ConnectionReuseStrategy;
+import org.apache.http.client.CredentialsProvider;
+import org.apache.http.client.HttpClient;
+import org.apache.http.impl.client.LaxRedirectStrategy;
+import org.apache.http.params.HttpParams;
+
+@SuppressWarnings("deprecation")
+public interface HttpExecutor extends HttpClient {
+
+    void setParams(HttpParams httpParams);
+
+    void setRedirectStrategy(LaxRedirectStrategy laxRedirectStrategy);
```

Comment:
I'm not convinced we want to tie this interface to the `org.apache.http` classes. I was assuming we'd make it more general. And that we would not extend `HttpClient`, which would presumably make it a lot harder to supply implementations which do not use `org.apache.http`.